### ISSUE TITLE
sql: SHOW QUERIES should scrub BACKUP and RESTORE access tokens

### ISSUE THREAD

**Issue by benesch** [cockroachdb/cockroach#17420]
`BACKUP` and `RESTORE` queries can include sensitive cloud provider access tokens in the URL query string. We scrub these from `SHOW JOBS`, but they appear in `SHOW QUERIES`.

```
> SHOW JOBS;
-[ RECORD 1 ]
id                 | 267734287600386048
type               | RESTORE
description        | RESTORE tpch.* FROM 'azure://backup-test/benchmarks/tpch/scalefactor-10' WITH OPTIONS ('into_db'='tpch_golden')
username           | root
descriptor_ids     | {59,60,61,62,63,64,57,58}
status             | running
created            | 2017-08-03 16:07:00.292524+00:00
started            | 2017-08-03 16:07:00.434558+00:00
finished           | NULL
modified           | 2017-08-03 16:07:32.578871+00:00
fraction_completed | 0.025830257683992386
error              | 

> SHOW QUERIES
-[ RECORD 42 ]
query_id         | 14d761b2807339860000000000000001
node_id          | 1
username         | root
start            | 2017-08-03 16:07:00.038439+00:00
query            | RESTORE tpch.* FROM 'azure://backup-test/benchmarks/tpch/scalefactor-10?AZURE_ACCOUNT_NAME=cockroachbackuptest&AZURE_ACCOUNT_KEY=base64encodedsecret/==' WITH OPTIONS ('into_db'='tpch_golden')
client_address   | 127.0.0.1:35804
application_name | 
distributed      | false
phase            | executing
```

Not sure what the best way to hook in to `SHOW QUERIES` is. Another plan hook-style solution where ccl registers a query scrubbing callback?

/cc @danhhz @knz

**Comment by knz** [comment]
I want to understand why this information needs to be scrubbed, given that the SHOW statements are restrcited to the root user or the user who issued the query?

**Comment by danhhz** [comment]
I guess? Seems like a pretty dangerous game to play with your cloud storage credentials

**Comment by knz** [comment]
If you let an untrusted party log in to your cockroachdb cluster as root, you have bigger problems than just with the cloud storage...

**Comment by knz** [comment]
As I was explaining to nikhil, `CREATE USER` has the same problem.

**Comment by benesch** [comment]
Fixed by 666102a in #17786, which sidesteps the problem by hiding `BACKUP`/`RESTORE`/`ALTER` queries from `SHOW QUERIES`.
